src/cockroach_db/mod.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::BTreeMap;
+
+use testcontainers::{ core::WaitFor, Image, ImageArgs };
+
+const DEFAULT_IMAGE_NAME: &str = "cockroachdb/cockroach";
+const DEFAULT_IMAGE_TAG: &str = "latest-v23.2";
+
+/// Module to work with [`Cockroach DB`] inside of tests.
+///
+/// This module is based on the official [`Cockroach docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers::clients;
+/// use testcontainers_modules::cockroach_db;
+///
+/// let docker = clients::Cli::default();
+/// let cockroach = docker.run(cockroach_db::CockroachDb::default());
+///
+/// let http_port = cockroach.get_host_port_ipv4(26257);
+///
+/// // do something with the started cockroach instance..
+/// ```
+///
+/// [`Cockroach`]: https://www.cockroachlabs.com/
+/// [`Cockroach docker image`]: https://hub.docker.com/r/cockroachdb/cockroach
+#[derive(Debug)]
+pub struct CockroachDb {
+    name: String,
+    tag: String,
+    env_vars: BTreeMap<String, String>,
+}
+
+impl Default for CockroachDb {
+    fn default() -> Self {
+        CockroachDb::new(DEFAULT_IMAGE_NAME.to_string(), DEFAULT_IMAGE_TAG.to_string())
+    }
+}
+
+impl CockroachDb {
+    fn new(name: String, tag: String) -> Self {
+        CockroachDb {
+            name,
+            tag,
+            env_vars: Default::default(),
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct CockroachDbArgs;
+
+impl ImageArgs for CockroachDbArgs {
+    fn into_iterator(self) -> Box<dyn Iterator<Item = String>> {
+        Box::new(vec!["start-single-node".to_owned(), "--insecure".to_owned()].into_iter())
```

Comment:
I guess these arguments can't be controlled with env-variables?

In that case I'd suggest to expose `insecure` as an option of arguments. To provide some control over the behavior

What alternatives are there instead of `start-single-node`?

Reply:
Correct, those can't, as far I know or can find, be set via env vars. The documentation for cockroach command-line script can be found [here](https://www.cockroachlabs.com/docs/stable/cockroach-commands). The alternative to `start-single-node` is typically going to be `start`, which starts the node as part of a cluster. I think the common use-case for testcontainers is single-node, simple setup, which was why I hard-coded these. I can impl `Default` for these args perhaps, and expose a `new` that takes a `Vec<String>` in case someone wants to fire up a cluster with TLS enabled.